### PR TITLE
Add code lens extension (for Rails)

### PR DESCRIPTION
**Pull request by andyw8**
Based on https://github.com/Shopify/ruby-lsp-rails/pull/59

Reminder: Ensure @adisonlampert is credited in the release notes.

This depends on https://github.com/Shopify/vscode-ruby-lsp/pull/639

TODO:
- [x] Get working with Test Explorer
- [x] Verify Debug CodeLens
- [x] Update docs
- [x] Add screenshot gif
- [x] Specify minimum Ruby LSP version (needs https://github.com/Shopify/ruby-lsp/pull/734 to be released). Also remove the Gemfile entry.
- [x] Update types (avoid `T.unsafe`)

### Testing

```
  gem "ruby-lsp", github: "Shopify/ruby-lsp"
  gem "ruby-lsp-rails", github: "Shopify/ruby-lsp-rails", branch: "andyw8/add-code-lens-extension"
```

### CODE REVIEW

File: test/ruby_lsp_rails/code_lens_test.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        # The first 3 responses are for the test class.
+        # The last 3 are for the test declaration.
+        assert_equal(6, response.size)
+        assert_match("Run", response[3].command.title)
+        assert_equal("bin/rails test ../../../../../../../../fake.rb:2", response[3].command.arguments[2])
+        assert_match("Run In Terminal", response[4].command.title)
+        assert_match("Debug", response[5].command.title)
+
+      ensure
+        RubyLsp::Requests::Hover.listeners.clear
```
[the remark this answers is not in the document]
Comment (andyw8) [reply test/ruby_lsp_rails/code_lens_test.rb]: We don't actually need this line, I've removed it.

File: lib/ruby_lsp/ruby_lsp_rails/code_lens.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      ::RubyLsp::Requests::CodeLens.add_listener(self)
+
+      sig { override.returns(ResponseType) }
+      attr_reader :response
+
+      sig { params(uri: String, emitter: EventEmitter, message_queue: Thread::Queue).void }
+      def initialize(uri, emitter, message_queue)
+        @response = T.let([], ResponseType)
+        @visibility = T.let("public", String)
+        @prev_visibility = T.let("public", String)
```
Comment (egiurleo) [inline lib/ruby_lsp/ruby_lsp_rails/code_lens.rb:51]: Sorry if I'm missing something here -- do these values ever change? And does `@prev_visibility` ever get used?

Comment (andyw8) [reply lib/ruby_lsp/ruby_lsp_rails/code_lens.rb:51]: That may have been accidentally removed in a refactoring, but probably means we are missing a test, I'll look into.

**Review by egiurleo — APPROVED** [review]
🎉 🎉 🎉  Yay this is so cool!

File: lib/ruby_lsp/ruby_lsp_rails/code_lens.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Rails
+    # ![CodeLens demo](../../code_lens.gif)
+    #
+    # This feature adds several CodeLens features for Rails applications using the built-in test framework:
```
Comment (vinistock) [inline lib/ruby_lsp/ruby_lsp_rails/code_lens.rb:8]: Let's say Active Support test cases.

File: lib/ruby_lsp/ruby_lsp_rails/code_lens.rb [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    # - Run tests in the VS Terminal
+    # - Run tests in the VS Code Test Explorer
+    # - Debug tests
+    #
+    # The
+    # [code lens](https://microsoft.github.io/language-server-protocol/specification#textDocument_codeLens)
+    # request informs the editor of runnable commands such as tests
+    #
+    # Example:
```
Comment (vinistock) [inline lib/ruby_lsp/ruby_lsp_rails/code_lens.rb:17]:
```suggestion
    # # Example
```

File: lib/ruby_lsp/ruby_lsp_rails/code_lens.rb [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    #
+    # For the following code, Code Lenses will be added above the class definition above each test method.
+    #
+    # ```ruby
+    # Run
+    # class HelloTest < ActiveSupport::TestCase
+    #   test "outputs hello" do
```
Comment (vinistock) [inline lib/ruby_lsp/ruby_lsp_rails/code_lens.rb:24]:
```suggestion
    #   test "outputs hello" do # <- Will show code lenses for executing this test
```

File: sorbet/rbi/gems/ruby-lsp@0.6.0.rbi
Comment (vinistock) [inline sorbet/rbi/gems/ruby-lsp@0.6.0.rbi:1]: Update the RBI to be v0.6.2.

**Review by vinistock — APPROVED** [review]
Vini & Stan

**Pull request merged**